packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -3192,6 +3292,86 @@ class BaseModelSqlv2 {
     }
     return data;
   }
+
+  private _convertDateFormat(
+    dateTimeColumns: Record<string, any>[],
+    d: Record<string, any>,
+    isXcdbBase: BoolType,
+  ) {
+    if (d) {
+      for (const col of dateTimeColumns) {
+        if (d[col.title]) {
+          let keepLocalTime = true;
+
```

Comment:
You can avoid deeply nested if/for

In the above scenario, it could be

```js
if(!d)
   return d;
for (const col of dateTimeColumns) {
  if (!d[col.title]) 
     continue;
 let keepLocalTime = true;
```

This way code will be more readable